R/join_key.R

```diff
@@ -0,0 +1,83 @@
+#' Create a relationship between a pair of datasets
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' @details `join_key()` will create a relationship for the variables on a pair
+#' of datasets.
+#'
+#' @param dataset_1 (`character(1)`) dataset name.
+#' @param dataset_2 (optional `character(1)`) other dataset name. In case it is omitted, then it
+#' will create a primary key for `dataset_1`.
+#' @param keys (optionally named `character`) where `names(keys)` are columns in `dataset_1`
+#' with relationship to columns of `dataset_2` given by the elements in `keys`.
+#' If `names(keys)` is `NULL` then the same column names are used for both `dataset_1`
+#' and `dataset_2`.
+#'
+#' @return object of class `join_key_set` to be passed into `join_keys` function.
+#'
+#' @seealso [join_keys()]
+#'
+#' @export
+#'
+#' @examples
+#' join_key("d1", "d2", c("A"))
+#' join_key("d1", "d2", c("A" = "B"))
+#' join_key("d1", "d2", c("A" = "B", "C"))
+join_key <- function(dataset_1, dataset_2 = dataset_1, keys) {
+  checkmate::assert_string(dataset_1)
+  checkmate::assert_string(dataset_2)
+  checkmate::assert_character(keys, any.missing = FALSE)
+
+  if (length(keys) > 0) {
+    if (is.null(names(keys))) {
+      names(keys) <- keys
+    }
+
+    keys <- trimws(keys)
+    names(keys) <- trimws(names(keys))
```

Comment:
I don't agree with this change as it would let untrimmed original names to pass

``` r
keys <- c(" name " = " value ")

keys <- trimws(keys)
if (is.null(names(keys))) {
  names(keys) <- keys
}
names(keys) == "name"
#> [1] FALSE
```

Reply:
You're right.